sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/indexes/FieldBuilder.java

```diff
@@ -38,28 +43,26 @@
         SUPPORTED_NONE_PARAMETERIZED_TYPE.put(String.class, SearchFieldDataType.STRING);
         SUPPORTED_NONE_PARAMETERIZED_TYPE.put(Date.class, SearchFieldDataType.DATE_TIME_OFFSET);
         SUPPORTED_NONE_PARAMETERIZED_TYPE.put(OffsetDateTime.class, SearchFieldDataType.DATE_TIME_OFFSET);
-//        SUPPORTED_NONE_PARAMETERIZED_TYPE.put(PointGeometry.class, SearchFieldDataType.GEOGRAPHY_POINT);
+        SUPPORTED_NONE_PARAMETERIZED_TYPE.put(PointGeometry.class, SearchFieldDataType.GEOGRAPHY_POINT);
     }
 
-    private static final List<Class<?>> UNSUPPORTED_TYPES = Arrays.asList(Byte.class,
-        CharSequence.class,
-        Character.class,
-        char.class,
-        Float.class,
-        float.class,
-        Short.class,
-        short.class);
+    private static final List<Class<?>> UNSUPPORTED_TYPES = Arrays.asList(byte.class, Byte.class,
+        CharSequence.class, Character.class, char.class,
+        Float.class, float.class,
+        Short.class, short.class);
 
     /**
      * Creates a collection of {@link SearchField} objects corresponding to the properties of the type supplied.
      *
      * @param modelClass The class for which fields will be created, based on its properties.
+     * @param serializer Optional serializer which allow to use customized serializer library. Default to take Jackson
+     * serialization.
      * @param <T> The generic type of the model class.
      * @return A collection of fields.
      */
-    public static <T> List<SearchField> build(Class<T> modelClass) {
+    public static <T> List<SearchField> build(Class<T> modelClass, PropertyNameSerializer serializer) {
```

Comment:
Could we call this method `getSearchFields`? `build` feels analogous to a builder pattern.

Reply:
Then the API is different than .NET.